### ISSUE TITLE
TDD: imag_return regression guard

### ISSUE THREAD

**Issue by LSailer** [LSailer/Master-Thesis-3D-VLA#25]
## TDD Cycle 4: `imag_return` metric regression guard

**Feature:** `train_step` returns `imag_return` in metrics dict (regression guard)

**Spec ref:** `docs/specs/dreamerv3-objectnav-baseline.md` §7, Cycle 4

### Test (`tests/test_dreamerv3_shapes.py`)

```python
class TestImaginationMetrics:
    def test_imag_return_in_metrics(self, cfg, rng):
        """train_step must return imag_return metric."""
        from src.dreamerv3.agent import DreamerAgent
        agent = DreamerAgent(cfg, rng)
        B, T = cfg.batch_size, cfg.seq_len
        batch = {
            "obs": jnp.zeros((B, T, *cfg.obs_shape)),
            "actions": jnp.zeros((B, T), dtype=jnp.int32),
            "rewards": jnp.zeros((B, T)),
            "dones": jnp.zeros((B, T)),
            "is_first": jnp.zeros((B, T)),
        }
        metrics = agent.train_step(batch, rng)
        assert "imag_return" in metrics, "imag_return must be in train_step metrics"
        assert isinstance(metrics["imag_return"], float)
```

### Implementation

None expected — should pass immediately. If not, check `_train_behavior` returns `imag_return` key.

### Pass command
```bash
uv run pytest tests/test_dreamerv3_shapes.py::TestImaginationMetrics -v
```

### Commit message
`add imag_return regression guard test`

### Files
- `tests/test_dreamerv3_shapes.py` (modify)
